### ISSUE TITLE
Outline feature does not work with long page with many headings

### ISSUE THREAD

**Issue by kingfungdesign** [GorvGoyl/Notion-Boost-browser-extension#176]
Enabling the `outline` feature will cause the typing experience to drastically slow down in a page that has many togglable headings and more headings nested underneath the togglable heading. 

To reproduce:
1. Create a page that has many togglable headings and headings nested underneath them. 
2. The typing responsivness will keep decreasing as more headings are being added.

Browser: Microsoft Edge Beta

**Comment by kingmadwik** [comment]
yes this is troubling, i have 7 toggles and it drastically reduces the typing and scrolling experience overall i end up disabling outline feature, can you please fix it dev

**Comment by BlackiePiggy** [comment]
same here, I use it to write dairy, every toggle is a date, I meet the same problem when it comes to 30days.

**Comment by ciqujmeoh77** [comment]
you can try Notion Mate

https://handy-humidity-660.notion.site/Notion-Mate-f0d639e5f48d4aeabde46f1fabb66a82
